### PR TITLE
[opa] Pod Security Standards

### PR DESCRIPTION
**Pull request by yalosev**
Signed-off-by: Yuriy Losev <yuriy.losev@flant.com>

## Description
Realize pod security standards

## Why do we need it, and what problem does it solve?
Give clients simple way to follow common security standards. This PR realize two ready-to-use profiles which can be activated via namespace label

Let's take the pod definition:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: nginx-privileged
  namespace: testns
  labels:
    app: nginx-privileged
spec:
  containers:
    - name: nginx
      image: nginx
      securityContext:
        privileged: true
```

and simple ns:
```yaml
apiVersion: v1
kind: Namespace
metadata:
  labels:
    kubernetes.io/metadata.name: testns
  name: testns
```

Given pod usually can be easily created in the ns:
```bash
# kubectl create -f pod_priv.yaml
pod/nginx-privileged created
```

Let's mark our ns with the baseline security profile:
```bash
# kubectl label ns testns security.deckhouse.io/pod-policy=baseline
```

and then try to create pod again:
```bash
# kubectl create -f pod_priv.yaml
Error from server (Forbidden): error when creating "pod_priv.yaml": admission webhook "admission-policy-engine.deckhouse.io" denied the request: [d8-pod-security-baseline] Privileged container is not allowed: nginx, securityContext: {"privileged": true}
```

next, let's activate restricted profile:
```bash
# kubectl label ns testns security.deckhouse.io/pod-policy=restricted --overwrite
````

and try one more time:
```bash
# kubectl create -f pod_priv.yaml
Error from server (Forbidden): error when creating "pod_priv.yaml": admission webhook "admission-policy-engine.deckhouse.io" denied the request: [d8-pod-security-restricted] Pod nginx-privileged is attempting to run without a required securityContext/runAsNonRoot or securityContext/runAsUser != 0
[d8-pod-security-restricted] Privilege escalation container is not allowed: nginx
[d8-pod-security-baseline] Privileged container is not allowed: nginx, securityContext: {"privileged": true}
[d8-pod-security-restricted] Seccomp profile 'undefined' is not allowed for container 'nginx'. Found at: no explicit profile found. Allowed profiles: {"Localhost", "RuntimeDefault"}
[d8-pod-security-restricted] container <nginx> is not dropping all required capabilities. Container must drop all of ["ALL"] or "ALL"
```

as you can see, restricted profile has more security requirements.

---
If we try to create a deployment with an unsecured pod, pod will not be created and deployment will have a status:
```yaml
# kubectl -n testns get deployment nginx-deployment -o yaml

status:
  conditions:
  ...
    - lastTransitionTime: "2022-10-07T13:41:18Z"
    lastUpdateTime: "2022-10-07T13:41:18Z"
    message: |-
      admission webhook "admission-policy-engine.deckhouse.io" denied the request: [d8-pod-security-restricted] Pod nginx-deployment-6595874d85-fh7t5 is attempting to run without a required securityContext/runAsNonRoot or securityContext/runAsUser != 0
      [d8-pod-security-restricted] Privilege escalation container is not allowed: nginx
      [d8-pod-security-restricted] Seccomp profile 'undefined' is not allowed for container 'nginx'. Found at: no explicit profile found. Allowed profiles: {"Localhost", "RuntimeDefault"}
      [d8-pod-security-restricted] container <nginx> is not dropping all required capabilities. Container must drop all of ["ALL"] or "ALL"
    reason: FailedCreate
```

## Checklist
- [x] The code is covered by unit tests.
- [x] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: admission-policy-engine
type: feature
summary: |
  New module `admission-policy-engine` with realized [Pod Security Standards](https://kubernetes.io/docs/concepts/security/pod-security-standards/). 
  Security profiles could be activated by setting a label `security.deckhouse.io/pod-policy: restricted` or `security.deckhouse.io/pod-policy: baseline` to the desired Namespace. 
  Added Grafana dashboard `Application/Admission Policy Engine` and the `PodSecurityStandardsViolation` alert.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

File: modules/015-admission-policy-engine/hooks/install_constraint_templates.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2021 Flant JSC
```
Comment (RomanenkoDenys) [inline modules/015-admission-policy-engine/hooks/install_constraint_templates.go:2]:
```suggestion
Copyright 2022 Flant JSC
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @yalosev. I've started the workflow for label 'e2e/run/aws'...

**Review by RomanenkoDenys — APPROVED** [review]

**Review by apolovov — APPROVED** [review]

**Review by kirkonru — COMMENTED** [review]
A few minor corrections/paraphrases. The following files were reviewed:

- openapi/config-values.yaml
- openapi/doc-ru-config-values.yaml
- modules/docs directory

File: modules/015-admission-policy-engine/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+- `security.deckhouse.io/pod-policy=baseline`
+- `security.deckhouse.io/pod-policy=restricted`
+
+For example:
+
+```bash
+kubectl label ns my-namespace security.deckhouse.io/pod-policy=restricted
+```
+
+Will set the `Restricted` policy for all Pods in the Namespace `my-namespace`
```
[the remark this answers is not in the document]
Comment (z9r5) [reply modules/015-admission-policy-engine/docs/README.md]: Fixed

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/test'...

**Review by z9r5 — APPROVED** [review]

**Pull request merged**